python/tests/image/test_imageIO.py

```diff
@@ -72,71 +74,59 @@ def tearDownClass(cls):
         imgAsRow = imageIO.imageArrayToStruct(array)
-        smaller = imageIO._resizeFunction([4, 5])
+        imgAsPIL = PIL.Image.fromarray(obj=imageIO._reverseChannels(array)).resize((5,4))
+        smallerAry = imageIO._reverseChannels(np.asarray(imgAsPIL))
+        smaller = imageIO.createResizeImageUDF([4, 5]).func
         smallerImg = smaller(imgAsRow)
-        for n in imageIO.imageSchema.names:
+        np.testing.assert_array_equal(smallerAry,imageIO.imageStructToArray(smallerImg))
+        for n in ImageSchema.imageSchema['image'].dataType.names:
             smallerImg[n]
         self.assertEqual(smallerImg.height, 4)
         self.assertEqual(smallerImg.width, 5)
-
-        sameImage = imageIO._resizeFunction([imgAsRow.height, imgAsRow.width])(imgAsRow)
-        self.assertEqual(sameImage, sameImage)
-
-        self.assertRaises(ValueError, imageIO._resizeFunction, [1, 2, 3])
-
-    def test_imageArrayToStruct(self):
-        SparkMode = imageIO.SparkMode
-        # Check converting with matching types
-        height, width, chan = array.shape
-        imgAsStruct = imageIO.imageArrayToStruct(array)
-        self.assertEqual(imgAsStruct.height, height)
-        self.assertEqual(imgAsStruct.width, width)
-        self.assertEqual(imgAsStruct.data, array.tobytes())
-
-        # Check casting
-        imgAsStruct = imageIO.imageArrayToStruct(array, SparkMode.RGB_FLOAT32)
-        self.assertEqual(imgAsStruct.height, height)
-        self.assertEqual(imgAsStruct.width, width)
-        self.assertEqual(len(imgAsStruct.data), array.size * 4)
-
-        # Check channel mismatch
-        self.assertRaises(ValueError, imageIO.imageArrayToStruct, array, SparkMode.FLOAT32)
-
-        # Check that unsafe cast raises error
-        floatArray = np.zeros((3, 4, 3), dtype='float32')
-        self.assertRaises(ValueError, imageIO.imageArrayToStruct, floatArray, SparkMode.RGB)
+        self.assertRaises(ValueError, imageIO.createResizeImageUDF, [1, 2, 3])
+        sameImage = imageIO.createResizeImageUDF((imgAsRow.height, imgAsRow.width)).func(imgAsRow)
+        self.assertEqual(imgAsRow, sameImage)
+
+    def test_imageConversions(self):
+        """"
+        Test conversion image array <-> image struct
+        """
+        def _test(array):
+            height, width, chan = array.shape
+            imgAsStruct = imageIO.imageArrayToStruct(array)
+            self.assertEqual(imgAsStruct.height, height)
+            self.assertEqual(imgAsStruct.width, width)
+            self.assertEqual(imgAsStruct.data, array.tobytes())
+            imgReconstructed = imageIO.imageStructToArray(imgAsStruct)
+            np.testing.assert_array_equal(array,imgReconstructed)
+        for nChannels in (1,3,4):
+            # unsigned bytes
+            _test(np.random.randint(0, 256, (10, 11, nChannels), 'uint8'))
+            _test(np.random.random_sample((10,11,nChannels)).astype('float32'))
 
     def test_image_round_trip(self):
```

Comment:
let's remove this one since the tests directly above and below test what this is testing more clearly.